spec/lib/embed/viewer/model_viewer_spec.rb

```diff
@@ -8,19 +8,66 @@
   let(:purl) do
     instance_double(
       Embed::Purl,
-      contents: [
-        instance_double(Embed::Purl::Resource, three_dimensional?: true,
-                                               files: [instance_double(Embed::Purl::ResourceFile, title: 'obj1.glb', file_url: '//file/druid:abc123/obj1.glb', downloadable?: true)])
-      ],
+      contents:,
+      location_restriction: false,
+      embargoed?: false,
+      stanford_only_unrestricted?: false,
       druid: 'abc123'
     )
   end
+  let(:contents) do
+    [
+      instance_double(Embed::Purl::Resource, three_dimensional?: true,
+                                             files: [instance_double(Embed::Purl::ResourceFile, title: 'obj1.glb', file_url: '//file/druid:abc123/obj1.glb', no_download?: false)])
+    ]
+  end
   let(:request) { instance_double(Embed::Request, purl_object: purl) }
 
   describe '#three_dimensional_files' do
-    it 'returns the full file URL for the PDFs in an object' do
-      expect(viewer.three_dimensional_files.length).to eq 1
-      expect(viewer.three_dimensional_files.first).to match(%r{/file/druid:abc123/obj1\.glb$})
+    context 'with downloadable files' do
+      it 'returns the full file URL for the PDFs in an object' do
+        expect(viewer.three_dimensional_files.length).to eq 1
+        expect(viewer.three_dimensional_files.first).to match(%r{/file/druid:abc123/obj1\.glb$})
+      end
+    end
+
+    context 'with no download' do
+      let(:contents) do
+        [
+          instance_double(Embed::Purl::Resource, three_dimensional?: true,
+                                                 files: [instance_double(Embed::Purl::ResourceFile, title: 'obj1.glb', file_url: '//file/druid:abc123/obj1.glb', no_download?: true)])
+        ]
+      end
+
+      it 'returns no files' do
+        expect(viewer.three_dimensional_files.length).to eq 0
+      end
+    end
+  end
+
+  describe 'message' do
```

Comment:
Should these tests move to authorization_spec.rb?. They seem to be about that class, not ModelViewer.

Reply:
Yeah sorry I wrote those before we moved everything to the authorization model

Reply:
Also based on our conversation on banners I think I need to go look at doing this in the file_auth component and not using this. Otherwise we are going to have to undo this.